edb/server/dbview/dbview.pyx

```diff
@@ -148,6 +160,73 @@ cdef class Database:
     def tenant(self):
         return self._index._tenant
 
+    def stop(self):
+        if self._cache_worker_task:
+            self._cache_worker_task.cancel()
+            self._cache_worker_task = None
+
+    async def cache_worker(self):
+        while True:
+            try:
+                await self._cache_worker()
+            except Exception as ex:
+                debug.dump(ex)
+                metrics.background_errors.inc(
+                    1.0, self.tenant._instance_name, "cache_worker"
+                )
+
+    async def _cache_worker(self):
+        added_since_signal = 0
+
+        i = 0
+        while True:
+            # First, handle any evictions
+            keys = []
+            while self._eql_to_compiled.needs_cleanup():
+                query_req, unit_group = self._eql_to_compiled.cleanup_one()
+                if len(unit_group) == 1 and unit_group.cache_state == 1:
+                    keys.append(query_req.get_cache_key())
+                unit_group.cache_state = CacheState.Evicted
+            if keys:
+                await self.tenant.evict_query_cache(self.name, keys)
+
+            # Now, populate the cache
+            query_req, units = await self._cache_queue.get()
+            # TODO: Should we do any sort of error handling here?
+            if (
+                len(units) == 1
+                and units[0].cache_sql
+                and units.cache_state == CacheState.Pending
+            ):
+                g = execute.build_cache_persistence_units([(query_req, units)])
+                conn = await self.tenant.acquire_pgcon(self.name)
+                try:
+                    await g.execute(conn, self)
+                finally:
+                    self.tenant.release_pgcon(self.name, conn)
+
+                units.cache_state = CacheState.Present
+
+                added_since_signal += 1
+
+            # Only signal query-cache-changes if we are quiescent or
+            # we have a lot that have updated since the last send.
+            # TODO: Is this good enough? Should we be smarter and try
+            # to rate-limit more and/or include a detailed payload.
+            await asyncio.sleep(0)
+            if (
+                added_since_signal
+                and (
+                    self._cache_queue.empty()
+                    or added_since_signal > 100
```

Comment:
Yeah I planned to do something along those lines, but I wanted to see if the aarch64 trouble recurred after these changes without it, but the aarch64 builds are too broken because of spot instance trouble to know yet really

Reply:
Alright I implemented something. Let me know if it basically matches what you were thinking

Reply:
I read the code a few times and I think it's good. I want @fantix to thoroughly read it too.